src/pretix/presale/forms/renderers.py

```diff
@@ -1,5 +1,38 @@
 from bootstrap3.renderers import FieldRenderer
+from bootstrap3.text import text_value
 from bootstrap3.utils import add_css_class
+from django.forms import CheckboxInput
+from django.forms.utils import flatatt
+from django.utils.html import format_html
+from django.utils.safestring import mark_safe
+from django.utils.translation import pgettext
+
+
+def render_label(content, label_for=None, label_class=None, label_title='', optional=False):
+    """
+    Render a label with content
+    """
+    attrs = {}
+    if label_for:
+        attrs['for'] = label_for
+    if label_class:
+        attrs['class'] = label_class
+    if label_title:
+        attrs['title'] = label_title
+
+    if text_value(content) == '&#160;':
+        # Empty label, e.g. checkbox
+        attrs.setdefault('class', '')
+        attrs['class'] += ' label-empty'
+
+    builder = '<{tag}{attrs}>{content}{opt}</{tag}>'
+    return format_html(
+        builder,
+        tag='label',
+        attrs=mark_safe(flatatt(attrs)) if attrs else '',
+        opt=mark_safe('<i class="sr-only"> {}</i>'.format(pgettext('form', 'required'))) if not optional else '',
```

Comment:
Would a screen reader read "ZIP code (required)" differently than "ZIP code required"? When reading, it makes a difference and the () would be more appropriate – but when listening it's probably to fast to hear an actual difference even if the screen reader would put in a longer break?

Reply:
AFAIK the „()“ will be read out as „parenthesis“, which would sound rather odd. That’s why it uses an i-element, which should change the voice, so it is clear, that „required“ is an instruction and not part of the label.

Reply:
Perfect, disrecard my comment then